index/secondary_index_manager.cc

```diff
@@ -8,20 +8,25 @@
  * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
  */
 
+#include <optional>
+#include <ranges>
```

Comment:
>  index: function that returns the custom class of an index, set of sup…
> 
> …posted custom classes

This commit is lacking motivation. I'm reading this as the first commit and don't understand why you need to introduce those functions.

I believe it would be better to introduce those functions in the same commit where you use it. Then, it won't be necessary to provide motivation - it should be clear from the contents of the commit that those helper functions support the goal that the commit strives to achieve.